[All Hosts] (webviews) emphasize webview terms instead of brower terms (#4018)
OfficeDev/office-js-docs-pr: docs/concepts/browsers-used-by-office-web-add-ins.md

---
title: Browsers and webview controls used by Office Add-ins
description: Specifies how the operating system and Office version determine what webview is used by Office Add-ins.
ms.topic: concept-article
ms.date: 05/20/2023
ms.localizationpriority: medium
---

# Browsers and webview controls used by Office Add-ins

> Office Add-ins are web applications that are displayed using iFrames when running in Office on the web. In Office for desktop and mobile clients, Office Add-ins use an embedded browser control (also known as a webview). Add-ins also need a JavaScript engine to run the JavaScript. Both the embedded browser and the engine are supplied by a browser installed on the user's computer. In this article, "webview" refers to the combination of a webview control and a JavaScript engine.

Reviewer: ```suggestion
Office Add-ins are web applications that are displayed using iframes when running in Office on the web. In Office for desktop and mobile clients, Office Add-ins use an embedded browser control (also known as a webview). Add-ins also need a JavaScript engine to run the JavaScript. Both the embedded browser and the engine are supplied by a browser installed on the user's computer. In this article, "webview" refers to the combination of a webview control and a JavaScript engine.
```

| Office 2019 | Volume-licensed | Doesn't matter | Trident+ (Internet Explorer 11) |
| Office 2016 | Volume-licensed | Doesn't matter | Trident+ (Internet Explorer 11) |
| Office 2013 | Doesn't matter | Doesn't matter | Trident+ (Internet Explorer 11) |

<sup>1</sup> On Windows versions prior to Windows 11, the WebView2 control must be installed so that Office can embed it. It's installed with perpetual Office 2021 or later; but it isn't automatically installed with Microsoft Edge. If you have an earlier version of perpetual Office, use the instructions for installing the control at [Microsoft Edge WebView2 / Embed web content ... with Microsoft Edge WebView2](https://developer.microsoft.com/microsoft-edge/webview2/).

<sup>2</sup> When you use either EdgeHTML or WebView2, the Windows Narrator (sometimes called a "screen reader") reads the `<title>` tag in the page that opens in the task pane. In Trident+, the Narrator reads the title bar of the task pane, which comes from the **\<DisplayName\>** value in the add-in's manifest.

> <sup>3</sup> If your add-in includes the **\<Runtimes\>** element in the manifest, then it will not use EdgeHTML. If the conditions for using WebView2 are met, then the add-in uses that browser. Otherwise, it uses Trident+. For more information, see [Runtimes](/javascript/api/manifest/runtimes).

Reviewer: ```suggestion
<sup>3</sup> If your add-in includes the **\<Runtimes\>** element in the manifest, then it won't use EdgeHTML. If the conditions for using WebView2 are met, then the add-in uses that browser. Otherwise, it uses Trident+. For more information, see [Runtimes](/javascript/api/manifest/runtimes).
```

<sup>2</sup> On Windows versions prior to Windows 11, the WebView2 control must be installed so that Office can embed it. It's installed with Microsoft 365, Version 2101 or later, but it isn't automatically installed with Microsoft Edge. If you have an earlier version of Microsoft 365, use the instructions for installing the control at [Microsoft Edge WebView2 / Embed web content ... with Microsoft Edge WebView2](https://developer.microsoft.com/microsoft-edge/webview2/). On Microsoft 365 builds before 16.0.14326.xxxxx, you must also create the registry key **HKEY_CURRENT_USER\SOFTWARE\Microsoft\Office\16.0\WEF\Win32WebView2** and set its value to `dword:00000001`.

<sup>3</sup> When you use either EdgeHTML or WebView2, the Windows Narrator (sometimes called a "screen reader") reads the `<title>` tag in the page that opens in the task pane. In Trident+, the Narrator reads the title bar of the task pane, which comes from the **\<DisplayName\>** value in the add-in's manifest.

> <sup>4</sup> If your add-in includes the **\<Runtimes\>** element in the manifest, then it will not use EdgeHTML. If the conditions for using WebView2 are met, then the add-in uses that browser. Otherwise, it uses Trident+ regardless of the Windows or Microsoft 365 version. For more information, see [Runtimes](/javascript/api/manifest/runtimes).

Reviewer: ```suggestion
<sup>4</sup> If your add-in includes the **\<Runtimes\>** element in the manifest, then it won't use EdgeHTML. If the conditions for using WebView2 are met, then the add-in uses that browser. Otherwise, it uses Trident+ regardless of the Windows or Microsoft 365 version. For more information, see [Runtimes](/javascript/api/manifest/runtimes).
```

Trident+ doesn't support JavaScript versions later than ES5. If any of your add-in's users have platforms that use Trident+, then to use the syntax and features of ECMAScript 2015 or later, you have two options.

- Write your code in ECMAScript 2015 (also called ES6) or later JavaScript, or in TypeScript, and then compile your code to ES5 JavaScript using a compiler such as [babel](https://babeljs.io/) or [tsc](https://www.typescriptlang.org/index.html).
- Write in ECMAScript 2015 or later JavaScript, but also load a [polyfill](https://en.wikipedia.org/wiki/Polyfill_(programming)) library such as [core-js](https://github.com/zloirock/core-js) that enables IE to run your code.

For more information about these options, see [Support older Microsoft webviews and Office versions](../develop/support-ie-11.md).

Also, Trident+ doesn't support some HTML5 features such as media, recording, and location. To learn more, see [Determine the webview the add-in is running in at runtime](../develop/support-ie-11.md#determine-the-webview-the-add-in-is-running-in-at-runtime).

## Troubleshoot EdgeHTML and WebView2 (Microsoft Edge) issues

### Service Workers aren't working

> Office Add-ins don't support Service Workers when the EdgeHTML is used. They're supported with WebView2.

Reviewer: ```suggestion
Office Add-ins don't support Service Workers when EdgeHTML is used. They're supported with WebView2.
```
